lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -528,19 +546,82 @@ getLegacyTransaction tid = do
     let inps = mempty
     pure $ Tx tid inps outs
 
+
+putWitness :: TxWitness -> Put
+putWitness (TxWitness bytes) = putByteString bytes
+
 putSignedTx :: [(TxIn, Coin)] -> [TxOut] -> [TxWitness] -> Put
 putSignedTx inputs outputs witnesses = do
-    putTx inputs outputs
+    putIO inputs outputs
```

Comment:
I would probably change name here as it suggests that something is about `IO` ...